sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/SharedAccessAuthorizationRule.cs

```diff
@@ -268,9 +268,9 @@ public override bool Equals(AuthorizationRule other)
         private static string GenerateRandomKey()
         {
             byte[] key256 = new byte[32];
-            using (var rngCryptoServiceProvider = new RNGCryptoServiceProvider())
+            using (var rng = RandomNumberGenerator.Create())
```

Comment:
Does this need to be created/used in a thread-safe way? https://devblogs.microsoft.com/pfxteam/getting-random-numbers-in-a-thread-safe-way/

Reply:
I found a lot of docs and articles talking about System.Random not being thread safe, such as [System.Random remarks](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-random#thread-safety).  I couldn't find a lot about RandomNumberGenerator but it sounds like instances created from RandomNumberGenerator.Create are thread safe: https://github.com/dotnet/dotnet-api-docs/issues/3741#issuecomment-718989978

Will keep looking for more information.

Reply:
Some more context I'm looking at - https://github.com/dotnet/runtime/issues/40169  / https://github.com/dotnet/runtime/pull/52495

Reply:
I might change this to be RandomNumberGenerator for .NET6+ only, it seems like it may not be thread-safe on .NET Framework, but it is on .NET Core

Reply:
I think we're good with the implementation is as-is for thread safety:
  
  - You cannot create an instance of the class directly, only via the  `Create` factory, which returns a singleton instance.   _([src](https://source.dot.net/#System.Security.Cryptography/System/Security/Cryptography/RandomNumberGenerator.cs,828597225647bdbb))_

  - The implementation, `RandomNumberGeneratorImplementation`, is sealed and has no class-level state.   Operations are self-contained.  _([src](https://source.dot.net/#System.Security.Cryptography/System/Security/Cryptography/RandomNumberGeneratorImplementation.cs,c96647a26999370a))_

  - The [docs](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.randomnumbergenerator?view=netframework-4.6.2#remarks) for the class state that `Create` spawns an RNGCryptoServiceProvider on down-level frameworks.   This can be traced back to [the .NET Framework implementation](https://referencesource.microsoft.com/#mscorlib/system/security/cryptography/randomnumbergenerator.cs,7741fdcbc20bd18c) calling into `CryptoConfig` with its own type name, which is [special-cased to RNGCryptoServiceProvider](https://referencesource.microsoft.com/#mscorlib/system/security/cryptography/cryptoconfig.cs,258).


I'll sign off on this as one of the Service Bus owners.